Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml

```diff
@@ -0,0 +1,170 @@
+                AzureDeploymentID,
+                Role
+      };
+  let Sysmon3_WindowsEvent=(disabled:bool=false){
+        WindowsEvent
+        | where not(disabled) 
+        | where Provider == "Microsoft-Windows-Sysmon" and EventID == 3
+        | extend
+                SourceIp = tostring(EventData.SourceIp),
+                DestinationIp = tostring(EventData.DestinationIp),
+                DstHostname = tostring(EventData.DestinationHostname),
+                SrcHostname = tostring(EventData.SrcHostname),
+                RuleName = tostring(EventData.RuleName),
+                UtcTime = todatetime(EventData.UtcTime),
+                ProcessId = tostring(EventData.ProcessId),
+                Image = tostring(EventData.Image),
+                User = tostring(EventData.User),
+                Protocol = tostring(EventData.Protocol),
+                Initiated = tobool(EventData.Initiated), // Initiated indicates the process initiated a connection (meaning outbound)
+                SourceIsIpv6 = tobool(EventData.SourceIsIpv6),
+                SourcePort = toint(EventData.SourcePort),
+                SourcePortName = tostring(EventData.SourcePortName),
+                DestinationIsIpv6 = tobool(EventData.DestinationIsIpv6),
+                DestinationPort = toint(EventData.DestinationPort),
+                DestinationPortName = tostring(EventData.DestinationPortName)
+        | parse EventData.ProcessGuid with "{" ProcessGuid "}"
+        | project-away EventData
+        | project-away
+                       Provider,
+                       Channel,
+                       Task,
+                       Data,
+                       RawEventData,
+                       EventOriginId
+      };
+  union isfuzzy=true Sysmon3_Event,Sysmon3_WindowsEvent
+        | extend
+                SrcUsernameType = iff(not(Initiated), "Windows", ""),
+                SrcUsername = iff(not(Initiated), tostring(User), ""),
+                SrcProcessId = iff(not(Initiated), tostring(ProcessId), ""),
+                SrcProcessGuid = iff(not(Initiated), ProcessGuid, ""),
+                SrcProcessName = iff(not(Initiated), tostring(Image), ""),
+                SrcAppName = iff(not(Initiated), tostring(Image), ""),
+                SrcAppType = iff(not(Initiated), 'Process', ""),
+                DstUsernameType = iff(Initiated, "Windows", ""),
+                DstUsername = iff(Initiated, tostring(User), ""),
+                DstProcessId = iff(Initiated, tostring(ProcessId), ""),
+                DstProcessGuid = iff(Initiated, ProcessGuid, ""),
+                DstProcessName = iff(Initiated, tostring(Image), ""),
+                DstAppName = iff(Initiated, tostring(Image), ""),
+                DstAppType = iff(Initiated, 'Process', "")
+        | project-away  ProcessId, ProcessGuid, Image
+        | project-rename 
+                EventStartTime = UtcTime,
+                Dvc = Computer,
+                IpAddr = SourceIp,
```

Comment:
While correct, the lines below are confusing and hard to maintain. You should rename SourceIp to SrcIpAddr, and DestinationIp to DstIpAddr and extend the others as aliases.

```
IpAddr = SourceIp,
Dst = DestinationIp,
```